insonmnia/miner/salesman.go

```diff
@@ -0,0 +1,345 @@
+		} else if !orderId.IsZero() {
+			m.checkOrder(ctx, plan)
+		} else if dealId.IsZero() && orderId.IsZero() {
+			m.placeOrder(ctx, plan)
+		}
+	}
+}
+
+func (m *Salesman) restoreState() error {
+	askPlans := m.state.AskPlans()
+	for _, plan := range askPlans {
+		m.resources.Consume(plan)
+	}
+	//TODO: restore tasks
+	return nil
+}
+
+func (m *Salesman) checkDeal(ctx context.Context, plan *sonm.AskPlan) {
+	m.log.Debugf("checking deal %s for ask plan %s and order %s",
+		plan.DealID.Unwrap().String(), plan.ID, plan.GetOrderID().Unwrap().String())
+	deal, err := m.eth.Market().GetDealInfo(ctx, plan.GetDealID().Unwrap())
+	if err != nil {
+		m.log.Warnf("could not get deal info for order %s, ask %s - %s",
+			plan.GetOrderID().Unwrap().String(), plan.ID, err)
+		// TODO: log, what else can we do?
+		return
+	}
+
+	if deal.Status == sonm.DealStatus_DEAL_CLOSED {
+		err = m.resources.Release(plan.ID)
+		if err != nil {
+			m.log.Warnf("could not release resources for deal %s, order %s, ask %s - %s",
+				plan.DealID.Unwrap().String(), plan.OrderID.Unwrap().String(), plan.ID, err)
+		}
+		plan.DealID = nil
+		plan.OrderID = nil
+		m.state.SaveAskPlan(plan)
+	}
+}
+
+func (m *Salesman) checkOrder(ctx context.Context, plan *sonm.AskPlan) {
+	//TODO: validate deal that it is ours
+	m.log.Infof("checking order %s for ask plan %s", plan.GetOrderID().Unwrap().String(), plan.ID)
+	order, err := m.eth.Market().GetOrderInfo(ctx, plan.GetOrderID().Unwrap())
+	if err != nil {
+		m.log.Warnf("could not get order info for order %s - %s", plan.GetOrderID().Unwrap().String(), err)
+		// TODO: log, what else can we do?
+		return
+	}
+
+	if !order.DealID.IsZero() {
+		deal, err := m.eth.Market().GetDealInfo(ctx, order.DealID.Unwrap())
+		if err != nil {
+			m.log.Warnf("could not get deal info for ID %s from market - %s", order.DealID.Unwrap().String(), err)
+			// TODO: log, what else can we do?
+			return
+		}
+		m.dealChan <- deal
+
+		plan.DealID = order.DealID
+
+		if err := m.state.SaveAskPlan(plan); err != nil {
+			m.log.Warnf("could not get save ask plan with deal %s - %s", order.DealID.Unwrap().String(), err)
+			// TODO: log, what else can we do?
+			return
+		}
+	}
+}
+
+func (m *Salesman) placeOrder(ctx context.Context, plan *sonm.AskPlan) {
+	benchmarks, err := m.hardware.ResourcesToBenchmarks(plan.GetResources())
+	if err != nil {
+		m.log.Warnf("could not get benchmarks for ask plan %s - %s", plan.ID, err)
+		return
+	}
+	net := m.hardware.Network
+	order := &sonm.Order{
+		OrderType:      sonm.OrderType_ASK,
+		OrderStatus:    sonm.OrderStatus_ORDER_ACTIVE,
+		AuthorID:       sonm.NewEthAddress(crypto.PubkeyToAddress(m.ethkey.PublicKey)),
+		CounterpartyID: plan.GetCounterparty(),
+		Duration:       uint64(plan.GetDuration().Unwrap().Seconds()),
+		Price:          plan.GetPrice().GetPerSecond(),
+		//TODO:refactor NetFlags in separqate PR
+		Netflags:      sonm.NetflagsToUint([3]bool{net.GetOverlay(), true, net.Incoming}),
```

Comment:
`net.GetOverlay()` returns bool?

Reply:
yep, I will use finally available GetOutbound and GetIncoming() for consistency